server/schemas/schemaMatcher.js

```diff
@@ -0,0 +1,72 @@
+var csv = require("csv");
+var Readable = require("stream").Readable;
+var schemaModel = require("./schemaModel");
+var helpers = require("../config/helpers");
+
+//turns an array into a query which can be sent to mongoDB
+//via mongoose in order to find a schema template with the same fields.
+var queryBuilder = function(array) {
+  var obj = {};
+  for (var i = 0; i < array.length; i++) {
+    if (!obj.hasOwnProperty(array[i].toString()) && array[i].length) {
```

Comment:
i'm having a really hard time understanding what is going on here. why the use of `hasOwnProperty` when obj doesn't seem to be part of a prototype chain? can the obj property values only be set to `true`? if so, why check the current value at all? What is the purpose of `array[i].length` check?